code/modules/virus2/effect/effect.dm

```diff
@@ -941,6 +941,25 @@ datum/disease2/effect/lubefoot/deactivate(var/mob/living/carbon/mob)
 				mob.apply_damage(10)
 
 
+/datum/disease2/effect/thick_blood
+	name = "Hyper-Fibrinogenesis"
+	stage = 3
+	var/skip = FALSE
+
+/datum/disease2/effect/thick_blood/activate(var/mob/living/carbon/mob,var/multiplier)
+	if(ishuman(mob))
+		var/mob/living/carbon/human/H = mob
+		if(H.species && (H.species.anatomy_flags & NO_BLOOD))	//Can't have clotting factors if you don't have blood!
+			skip = TRUE
+			return
+		if (H.reagents.get_reagent_amount(CLOTTING_AGENT) < 5)
+			H.reagents.add_reagent(CLOTTING_AGENT, 5)
```

Comment:
Clotting agent is metabolized whether you're bleeding or not, so wouldn't it be a waste to keep constantly putting more clotting agent into them even when they're not bleeding?
I think it would make more sense for it to check for bleeding, *then* check how much clotting agent they have and add more if necessary.